packages/openapi-generator/src/parser/operation-naming.spec.ts

```diff
@@ -0,0 +1,99 @@
+import { operationNameExtension } from '../extensions';
+import { OperationInfo } from './operation-info';
+import {
+  ensureUniqueOperationIds,
+  getOperationNameFromPatternAndMethod,
+  nameOperations
+} from './operation-naming';
+
+it('ensureUniqueOperationIds replaces duplicate names, while prioritizing original names', () => {
+  const uniqueOperations = ensureUniqueOperationIds([
+    { operation: { operationId: 'getX', summary: 'operation1' } },
+    { operation: { operationId: 'getX', summary: 'operation2' } },
+    { operation: { operationId: 'getX1', summary: 'operation3' } }
+  ] as OperationInfo[]);
+
+  expect(uniqueOperations.map(({ operation }) => operation)).toEqual([
+    { operationId: 'getX', summary: 'operation1' },
+    { operationId: 'getX1', summary: 'operation3' },
+    { operationId: 'getX2', summary: 'operation2' }
+  ]);
+});
+
+it('nameOperations adds retrieves initial names for operations', () => {
+  const uniqueOperations = nameOperations([
+    { operation: { operationId: 'getX', summary: 'operation1' } },
+    { method: 'get', pathPattern: '/x', operation: { summary: 'operation2' } },
+    {
+      operation: {
+        [operationNameExtension]: 'nameFromExtension',
+        summary: 'operation3'
+      }
+    }
+  ] as OperationInfo[]);
+
+  expect(uniqueOperations.map(({ operation }) => operation)).toEqual([
+    expect.objectContaining({
+      operationId: 'nameFromExtension',
+      summary: 'operation3'
+    }),
+    { operationId: 'getX', summary: 'operation1' },
+    { operationId: 'getX', summary: 'operation2' }
```

Comment:
duplicate `operationId`

Reply:
True, they are renamed in the next step. I know this is a little unusual, but I found it easier to understand when reading the code, than doing this in one step.